### PR TITLE
feat: Add heartbeat processing logic and implement GetTypes by reflection

### PR DESCRIPTION
**Pull request by DMwangnima**
#### Check the PR title.

- [ ] This PR title match the format: \<type\>(optional scope): \<description\>
- [ ] The description of this PR title is user-oriented and clear enough for others to understand.
- [ ] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional):

### CODE REVIEW

File: pkg/codec.go [hunk 1]
```diff
@@ -351,6 +375,11 @@ func processAttachments(decoder iface.Decoder, message remote.Message) error {
 	if err != nil {
 		return err
 	}
+	//
+	//if attachmentsRaw == nil || attachmentsRaw == "" {
+	//	attachmentsRaw = map[interface{}]interface{}{"interface": service.}
+	//}
```
Comment (felix021) [inline pkg/codec.go]: Will we use the code commented out later? Seems unlikely, the values have different types.

File: pkg/codec.go [hunk 2]
```diff
@@ -362,3 +391,21 @@ func processAttachments(decoder iface.Decoder, message remote.Message) error {
 
 	return fmt.Errorf("unsupported attachments: %v", attachmentsRaw)
 }
+
+func getTypes(data interface{}) (string, error) {
```
Comment (felix021) [inline pkg/codec.go]: I think it's better moved to pkg/dubbo/kitex.go, and we can maintain a cache with key = reflect.ValueOf(data)

File: pkg/dubbo/parameter.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	hessian "github.com/apache/dubbo-go-hessian2"
+)
+
+// GetParamsTypeList is copied from dubbo-go, it should be rewritten
```
Comment (felix021) [inline pkg/dubbo/parameter.go:103]: This piece of code is shorter than I expected, so we don't need to rewrite it.

Please add a formal statement after the license header explaining its source and our gratitude:

```
This source file has been replicated from the original Dubbo-Go project repository, and we extend our sincere appreciation to the Dubbo-Go development team for their valuable contribution.
```

Comment (jasondeng1997) [reply pkg/dubbo/parameter.go:103]: 参考dubbo-gohessian2这个库

**Review by jasondeng1997 — COMMENTED** [review]
LGTM

File: pkg/dubbo/parameter.go [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+	"strings"
+	"sync"
+	"time"
+
+	hessian "github.com/apache/dubbo-go-hessian2"
+)
+
+var (
+	typesMapLock sync.Mutex
+	typesMap     = make(map[reflect.Type]string)
```
Comment (felix021) [inline pkg/dubbo/parameter.go]: sync.Map 更适合现在这个case？

File: pkg/dubbo/parameter_test.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+		datum    []interface{}
+		expected func(t *testing.T, typesMap map[reflect.Type]string)
+	}{
+		{
+			desc: "same structs with basic Type",
+			datum: []interface{}{
+				&testStructA{Field: 1},
+				&testStructA{Field: 2},
+			},
+			expected: func(t *testing.T, typesMap map[reflect.Type]string) {
```
Comment (felix021) [inline pkg/dubbo/parameter_test.go]: 1. 这个 typesMap 和全局变量重名了，容易混淆
2. 单测依赖有副作用的结果（对全局变量的修改），还有优化空间。

如果想要优化的话可以这样：
1. 不要直接定义一个全局的 cache
2. 定义一个 type Cacher，New的时候指定一个 map 和 `Get(key) value` 方法
3. 定义全局变量 typeCacher = NewCacher(make(map...), getTypes)
4. 针对 Cacher 和 getTypes 分别写单测

Comment (felix021) [reply pkg/dubbo/parameter_test.go]: p.s. sync.SingleFlight 在 cache 场景会比较好用

https://pkg.go.dev/golang.org/x/sync/singleflight

File: pkg/dubbo/parameter.go [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+		if err != nil {
+			return "", err
+		}
+
+		return types, nil
+	})
+	if err != nil {
+		return "", err
+	}
+	tc.typesMap.Store(typ, typesRaw)
```
Comment (felix021) [inline pkg/dubbo/parameter.go]: 1. The `Store` operation should be called within the `Do` function.
2. if getParamsTypeList won't return an error in any case, we can safely ignore it.

**Review by felix021 — APPROVED** [review]

**Pull request merged**